plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+}
+
+func TestVolumeFilterFn(t *testing.T) {
+	db := connectTestDb()
+	dailyVolumeByDateQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, []string{})
+	if e != nil {
+		t.Log("could not make empty config volume query")
+		return
+	}
+
+	emptyFilter := &volumeFilter{
+		name:                   "volumeFilter",
+		configValue:            "",
+		baseAsset:              testNativeAsset,
+		quoteAsset:             testNativeAsset,
+		config:                 &VolumeFilterConfig{},
+		dailyVolumeByDateQuery: dailyVolumeByDateQuery,
+	}
+
+	testCases := []struct {
+		name      string
+		filter    *volumeFilter
+		dailyOTB  *VolumeFilterConfig
+		dailyTBB  *VolumeFilterConfig
+		op        *txnbuild.ManageSellOffer
+		wantOp    *txnbuild.ManageSellOffer
+		wantError error
+		wantTBB   *VolumeFilterConfig
+	}{
+		{
+			name:      "failure - is selling",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
+			dailyTBB:  nil,
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.CreditAsset{}},
+			wantOp:    nil,
+			wantError: fmt.Errorf("error when running the isSelling check for offer '{Selling:{Code: Issuer:} Buying:{} Amount: Price: OfferID:0 SourceAccount:<nil>}': invalid assets, there are more than 2 distinct assets: sdexBase={native  }, sdexQuote={native  }, selling={ }, buying={}"),
+			wantTBB:   nil,
+		},
+		{
+			name:      "failure - invalid sell price in op",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
+			dailyTBB:  nil,
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}},
+			wantOp:    nil,
+			wantError: fmt.Errorf("could not convert price () to float: strconv.ParseFloat: parsing \"\": invalid syntax"),
+			wantTBB:   nil,
+		},
+		{
+			name:      "failure - invalid amount in op",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
+			dailyTBB:  nil,
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}, Price: "0.0"},
+			wantOp:    nil,
+			wantError: fmt.Errorf("could not convert amount () to float: strconv.ParseFloat: parsing \"\": invalid syntax"),
+			wantTBB:   nil,
+		},
+		{
+			name:      "failure - invalid amount in op",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
+			dailyTBB:  nil,
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}, Price: "0.0"},
+			wantOp:    nil,
+			wantError: fmt.Errorf("could not convert amount () to float: strconv.ParseFloat: parsing \"\": invalid syntax"),
+			wantTBB:   nil,
+		},
+		{
+			name:      "success - selling, no filter sell caps",
+			filter:    emptyFilter,
+			dailyOTB:  &VolumeFilterConfig{},
+			dailyTBB:  &VolumeFilterConfig{SellBaseAssetCapInBaseUnits: createFloatPtr(0.0), SellBaseAssetCapInQuoteUnits: createFloatPtr(0.0)},
+			op:        &txnbuild.ManageSellOffer{Buying: txnbuild.NativeAsset{}, Selling: txnbuild.NativeAsset{}, Price: "1.0", Amount: "100.0"},
```

Comment:
we should have a helper function called `makeManageSellOffer(price, volume)` so we don't have to create structs everywhere -- makes it a lot more readable

Reply:
better yet -- this should be a parameter to the testCase struct input and the test should construct the appropriate manage sell offer from these two fields of price and volume

Reply:
similarly, the inputs for otb and tbb should be:
- otbBaseCap
- otbQuoteCap
- tbbBaseCap
- tbbQuoteCap

this will make these test case structs a lot smaller and will contain less boilerplate